### PR TITLE
Enhance and decouple the ImageVector library

### PR DESCRIPTION
**Pull request by adracus**
**What this PR does / why we need it**:
The reading of ImageVectors is now required in more than just the Gardener. To allow better ease-of-use, this refactoring allows easy `Read`ing and `Merge`ing of image vectors. This can then be consumed in the Gardener extensions as noted in a todo in https://github.com/gardener/gardener-extensions/pull/28.

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by timuthy — COMMENTED** [review]
/lgtm

Since you structured it so nicely, it'd be nice to see some tests added, though.

**Comment by adracus** [comment]
@timuthy you're right, I implemented the tests now.

File: pkg/utils/imagevector/imagevector.go [hunk 1]
```diff
@@ -14,73 +14,81 @@
...
-	for _, image := range overwrite {
-		overwrittenImages[image.Name] = image
-	}
+	return Read(file)
+}
 
-	var out ImageVector
-	for _, image := range vector {
-		if overwritten, ok := overwrittenImages[image.Name]; ok {
-			if len(overwritten.Tag) == 0 && len(image.Tag) > 0 {
```
Comment (rfranzke) [inline pkg/utils/imagevector/imagevector.go:58]: Why is that deleted?

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: pkg/utils/imagevector/imagevector.go (hunk 1 above)
Comment (adracus) [reply pkg/utils/imagevector/imagevector.go:58]: It is not deleted it is moved to `Merge`

Comment (rfranzke) [reply pkg/utils/imagevector/imagevector.go:58]: I don't see it. Where is the special handling for the `Tag` property?

**Review by rfranzke — APPROVED** [review]
/lgtm
